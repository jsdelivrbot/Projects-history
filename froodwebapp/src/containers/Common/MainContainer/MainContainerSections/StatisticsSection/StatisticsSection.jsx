import React from 'react';
import PropTypes from 'prop-types';
import {
  statistics,
  statisticsItem
} from './StatisticsSection.scss';

const checkColor = value => (
  value > 0 ? 'green' : 'red'
);

const simpleSection = (column, key) => (
  <div key={ key } className={ statisticsItem }>
    <div>{ column.dollarSign ? `$${column.value || 0}` : column.value }</div>
    <div>{ column.description }</div>
  </div>
);

const dynamicSection = (column, key) => (
  <div key={ key } className={ statisticsItem }>
    <div style={ { color: checkColor(column.value) } }>
      {
        column.value > 0 ? `+${column.value}` : `-${column.value}`
      }
    </div>
    <div>{ column.description }</div>
  </div>
);

const Statistics = ({ stats }) => (
  <div className={ statistics }>
    { stats && stats.map((column, index) => (
      column.isDynamic
        ? dynamicSection(column, index)
        : simpleSection(column, index)
    ))}
  </div>
);

Statistics.propTypes = {
  stats: PropTypes.array,
};

export default Statistics;

