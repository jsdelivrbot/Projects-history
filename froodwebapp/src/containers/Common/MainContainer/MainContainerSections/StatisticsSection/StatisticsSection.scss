.statistics {
  text-align: center;
}

.statisticsItem {
  display: inline-block;
  margin: 0 2rem;
  div {
    &:nth-child(1) {
      font-size: 4rem;
    }
    &:nth-child(2) {
      font-size: 1rem;
    }
  }
}