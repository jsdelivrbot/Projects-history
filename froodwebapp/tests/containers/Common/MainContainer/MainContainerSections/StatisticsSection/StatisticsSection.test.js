import StatisticsSection from 'containers/Common/MainContainer/MainContainerSections/StatisticsSection/StatisticsSection';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, expect, it } = global;

describe('StatisticsSection', () => {
  const props = {
    stats: [{
      isDynamic: false,
      dollarSign: true,
      value: '12',
      description: 'Some description'
    }, {
      isDynamic: true,
      dollarSign: true,
      value: '12',
      description: 'Some description'
    }]
  };

  it('renders the StatisticsSection component', () => {
    const component = shallow(<StatisticsSection { ...props } />);

    expect(component.length).toEqual(1);
  });

  it('capturing Snapshot of StatisticsSection', () => {
    const component = shallow(<StatisticsSection { ...props } />);
    const view = toJson(component);
    const actual = expect(view);

    actual.toMatchSnapshot();
  });
});
