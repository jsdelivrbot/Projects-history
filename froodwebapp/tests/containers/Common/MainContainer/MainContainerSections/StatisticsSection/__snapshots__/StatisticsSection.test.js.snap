// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsSection capturing Snapshot of StatisticsSection 1`] = `
<div
  className="statistics"
>
  <div
    className="statisticsItem"
    key="0"
  >
    <div>
      $12
    </div>
    <div>
      Some description
    </div>
  </div>
  <div
    className="statisticsItem"
    key="1"
  >
    <div
      style={
        Object {
          "color": "green",
        }
      }
    >
      +12
    </div>
    <div>
      Some description
    </div>
  </div>
</div>
`;
